owner-extras/src/main/java/org/aeonbits/owner/loaders/MongoPropertiesLoader.java

```diff
@@ -0,0 +1,77 @@
+package org.aeonbits.owner.loaders;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Properties;
+
+import org.aeonbits.owner.loaders.Loader;
+import org.bson.Document;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+
+/**
+
+If you have a collection of key value pairs in mongo, you can use this class to pull values from it.
+
+	Loader loader = new MongoPropertiesLoader();
+	ConfigFactory.registerLoader(loader);
+	
+	CoreConfig cfg = ConfigCache.getOrCreate(CoreConfig.class);
+
+
+This class assumes that the collection of config values live in the "configuration" collection.  If different, set the property configurationCollectionName.
+
+This class needs a full mongo url in sources:
+
+	@Sources({"mongodb://user:password@master.hostname:11111/dbname","classpath:server.properties"})
+
+
+Code courtesy of Katasi LLC.
+
+*/
+
+
+public class MongoPropertiesLoader implements Loader {
+
+	private String configurationCollectionName = "configuration";
```

Comment:
I think that this information should be specified in the mongo connection url: remember that a single loader is used to load multiple sources, so it would be best your url to be formed as:

    mongodb://user:password@master.hostname:11111/dbname?collectionName=configuration

Or

    mongodb://user:password@master.hostname:11111/dbname#configuration

Then you may need to modify the parse the url and modify it before giving it to MongoClientURI() constructor.

For instance. In this way you don't need to modify the Loader with a setter method.